Tasks/KubernetesManifestV0/src/utils/DeploymentHelper.ts

```diff
@@ -137,3 +141,59 @@ function updateImagePullSecretsInManifestFiles(filePaths: string[], imagePullSec
 function isCanaryDeploymentStrategy(deploymentStrategy: string): boolean {
     return deploymentStrategy != null && deploymentStrategy.toUpperCase() == canaryDeploymentHelper.CANARY_DEPLOYMENT_STRATEGY.toUpperCase();
 }
+
+function checkPodStatus(kubectl: Kubectl, podName: string) {
+    const startTime = new Date();
+    const timeOut = 5 * 60 * 1000; // Timeout 5 min
+    let currentTime = new Date();
+    let podStatus;
+    while (currentTime.getTime() - startTime.getTime() < timeOut) {
+        tl.debug('Polling for pod status');
+        podStatus = getPodStatus(kubectl, podName);
+        if (podStatus.status && podStatus.phase) {
+            if (podStatus.phase !== 'Pending') {
+                break;
+            }
+        }
+        currentTime = new Date();
+    }
+    podStatus = getPodStatus(kubectl, podName);
+    switch (podStatus.phase) {
+        case 'Succeeded':
+        case 'Running':
+            if (checkIfAllContainersAreInReadyState(podStatus)) {
+                console.log(`pod/${podName} is successfully rolled out`);
+            }
+            break;
+        case 'Pending':
+            if (!checkIfAllContainersAreInReadyState(podStatus)) {
+                tl.warning(`pod/${podName} rollout status check timedout`);
+            }
+            break;
+        case 'Failed':
+            tl.error(`pod/${podName} rollout failed`);
+            break;
+        default:
+            tl.warning(`pod/${podName} rollout status: ${podStatus.status.phase}`);
+    }
+}
+
+function getPodStatus(kubectl: Kubectl, podName: string): any {
+    const podResult = kubectl.getResource('pod', podName);
+    utils.checkForErrors([podResult]);
+    return JSON.parse(podResult.stdout).status;
+}
+
+function checkIfAllContainersAreInReadyState(podStatus: any): boolean {
```

Comment:
Rename the function to something like = IsPodInReadyState

Reply:
Sure